### PR TITLE
[orc8r] Define a gRpc interface for swagger specs

### PR DESCRIPTION
**Pull request by AndyLKhuu**
Signed-off-by: Andy Lee Khuu <andykhuu@stanford.edu>

## Summary
As part of the effort to convert the merging process of SwaggerSpecs to runtime, we need to define a gRpc interface which allows us to poll registered services for the contents of their swagger specs. This PR defines said above gRpc interface for future use 

image
(Outline of overall architecture - blue outline is what the current PR covers)

## Test Plan
- [x] cd $MAGMA_ROOT/orc8r/cloud/docker && ./build.py -t ; noti

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: orc8r/cloud/go/obsidian/swagger/protos/swagger.proto [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+package magma.orc8r.obsidian.swagger;
+option go_package = "protos";
+
+// SwaggerClient provides a Swagger specification of the service's REST API.
+service SwaggerClient {
+ // GetSpec returns the Swagger specification of the service's REST API.
+ rpc GetSpec(GetSpecRequest) returns (GetSpecResponse) {}
+}
+
+message GetSpecRequest {
```
Comment (hcgatewood) [inline orc8r/cloud/go/obsidian/swagger/protos/swagger.proto]: Nit: for empty message definitions, we can just keep their defn to a single line

File: orc8r/cloud/go/obsidian/swagger/swagger_client.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package swagger
+
+// SwaggerClient is used to retrieve the Swagger specs of
+// registered services.
+type SwaggerClient interface {
```
Comment (hcgatewood) [inline orc8r/cloud/go/obsidian/swagger/swagger_client.go]: This file and definition is unnecessary -- the proto-generated code will have a definition for the client interface

File: orc8r/cloud/go/obsidian/swagger/protos/swagger.proto [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto3";
+package magma.orc8r.obsidian.swagger;
+option go_package = "protos";
+
+// SwaggerClient provides a Swagger specification of the service's REST API.
+service SwaggerClient {
```
Comment (hcgatewood) [inline orc8r/cloud/go/obsidian/swagger/protos/swagger.proto]: "Client" is useless/misleading description here. We're naming the service itself, which will have both a client and a service interface. Unless you have something specific in mind, let's go with `SwaggerSpec` as the service name

File: orc8r/cloud/go/obsidian/swagger/swagger_spec_client.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+	res, err := c.GetSpec(context.Background(), &protos.GetSpecRequest{})
+	if err != nil {
+		return "", err
+	}
+
+	return res.SwaggerSpec, nil
+}
+
+func (s *swaggerClient) GetSwaggerSpecClient() (protos.SwaggerSpecClient, error) {
```
Comment (hcgatewood) [inline orc8r/cloud/go/obsidian/swagger/swagger_spec_client.go]: - You're exporting a method that can only ever be called by the object itself -- but the object is private and the New method returns the iface. So something's wrong
- In this case, the answer to the issue is this method should be private

File: orc8r/cloud/go/obsidian/swagger/swagger_spec_client.go [hunk 5]
```diff
@@ -0,0 +1,59 @@
+/*
```
Comment (hcgatewood) [inline orc8r/cloud/go/obsidian/swagger/swagger_spec_client.go]: Rename this file to remote_spec.go

**Review by hcgatewood — CHANGES_REQUESTED** [review]

File: orc8r/cloud/go/obsidian/swagger/protos/swagger.proto (hunk 3 above)
Comment (AndyLKhuu) [reply orc8r/cloud/go/obsidian/swagger/protos/swagger.proto]: Thanks for the clarification, I was having trouble formalizing what to actually name these end points but the combined feedback in this PR makes it more clear.

File: orc8r/cloud/go/obsidian/swagger/remote_spec.go [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+	res, err := c.GetSpec(context.Background(), &protos.GetSpecRequest{})
+	if err != nil {
+		return "", err
+	}
+
+	return res.SwaggerSpec, nil
+}
+
+func (s *remoteSpec) getSwaggerSpecClient() (protos.SwaggerSpecClient, error) {
```
Comment (hcgatewood) [inline orc8r/cloud/go/obsidian/swagger/remote_spec.go]: Just call this `getClient` -- it's clear from the context and the return type what the type of the client is

**Review by hcgatewood — APPROVED** [review]
Lgtm. Just one naming nit. Message me when you fix and I'll merge

**Pull request merged**